File: garbage/solution/garbage-sub_1.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

int main() {
    // read input
    int N, W, H;
    cin >> N >> W >> H;
    vector<pair<int, int>> pts;
    vector<int> weights;
    for (int i = 0; i < N; i++) {
        int x, y, w;
        cin >> x >> y >> w;
        pts.push_back({x, y});
        weights.push_back(w);
    }

    // check all pairs of points as lower boundaries
    ll result = 0;
    for (int l = 0; l < N; l++) {
        for (int b = 0; b < N; b++) {
            ll value = 0;
            for (int i = 0; i < N; i++) {
                if (pts[i].first >= pts[l].first && pts[i].first < pts[l].first + W &&
                    pts[i].second >= pts[b].second && pts[i].second < pts[b].second + H) {
                        value += weights[i];
                }
            }
            result = max(result, value);
        }
    }
    cout << result << endl;
}

File: garbage/solution/garbage-sub_1.py
class point:
    def __init__(self, x, y, w):
        self.x = x
        self.y = y
        self.w = w

# read input
N, W, H = map(int, input().split())
pts = []
for i in range(N):
    x, y, w = map(int, input().split())
    pts.append(point(x, y, w))

# check all pairs of points as lower boundaries
result = 0
for l in range(N):
    for b in range(N):
        value = 0
        for i in range(N):
            if (pts[i].x >= pts[l].x and pts[i].x < pts[l].x + W and
                pts[i].y >= pts[b].y and pts[i].y < pts[b].y + H):
                    value += pts[i].w
        result = max(result, value)
print(result)
